### ISSUE TITLE
Whats the expected input to the losses and metrics?

### ISSUE THREAD

**Issue by lneukom** [allegro/allRank#58]
E.g. for **approxNDCGLoss**, what is **y_pred** and **y_true**?
- what is the range of the inputs?
- does a slate have to be in order? e.g. best ranked first?

From what I gathered:
- the order does not matter

And are all losses and metrics following this API?
Thanks!

**Comment by mhsyno** [comment]
Hello,

both **y_pred** and **y_true** are of type ```torch.Tensor``` with the same shape: ```[batch_size, slate_length]```
- **y_true** values are labels for each slate **in the original order**. The higher value the more relevant item is according in the context of a slate. Typically relevance takes integer (e. g. from 0 to 4 in [MSLR-WEB30K](https://www.microsoft.com/en-us/research/project/mslr/)) or binary values.
- **y_pred** values are real-valued scores from model which are used to produce new order of **y_true** according to these scores (descending sorting). They're also in the same order as **y_true**.

As for the question: 
> are all losses and metrics following this API?

In short: yes. For all the metrics and losses **y_pred** and **y_true** are mandatory arguments. Some of them have also additional arguments (most of them with default values specified) e. g. ```ats``` for metrics which specifies top n items of a slate taken into account while calculating a metric.
For some of the losses there are other mandatory arguements - when using [ordinal loss](https://github.com/allegro/allRank/blob/master/allrank/models/losses/ordinal.py#L25) you need to pass the number of ordinal values and for [pointwise rmse](https://github.com/allegro/allRank/blob/master/allrank/models/losses/pointwise.py) number of unique ground truth values is required.
Take a notice that some arguments of the loss functions can drastically change form of the function e. g. ```weighing_scheme``` for [lambdaLoss](https://github.com/allegro/allRank/blob/master/allrank/models/losses/lambdaLoss.py)

If in doubts I encourage to take a look at our real-life configs from the reproducibility guide: https://github.com/allegro/allRank/tree/master/reproducibility/configs

Best,
Mikołaj

**Comment by yunyu** [comment]
@mhsyno Super dumb question, why am I getting a negative/nonzero loss here?

```
approxNDCGLoss(
    y_pred=torch.tensor(
        [
            [0,0,1],
            [0,1,0],
            [1,0,0],
        ], dtype=torch.float32),
    y_true=torch.tensor(
        [
            [0,0,1],
            [0,1,0],
            [1,0,0],
        ], dtype=torch.float32),
)
```

Returns `tensor(-0.7443)`
